groups/bsl/bslstl/bslstl_variant.h

```diff
@@ -2828,18 +3025,17 @@ variant<HEAD, TAIL...>& variant<HEAD, TAIL...>::operator=(
     if (&lvalue != this) {
         if (this->d_type == lvalue.d_type) {
             if (!valueless_by_exception()) {
-                BloombergLP::bslstl::Variant_MoveAssignVisitor moveAssign(
-                                                 BSLS_UTIL_ADDRESSOF(d_union));
-                bsl::visit(moveAssign, lvalue);
+                BloombergLP::bslstl::Variant_MoveAssignVisitor<variant>
+                moveAssign(BSLS_UTIL_ADDRESSOF(lvalue));
+                BSLSTL_VARIANT_VISITID(void, variant, moveAssign, lvalue);
```

Comment:
```suggestion
                BSLSTL_VARIANT_VISITID(void, variant, moveAssign, *this);
```

Reply:
This requires a modification in the test suite. It should have caught this issue, but it looks like they never test the resulting value. Applied the suggested change locally so I can amend the tests at the same time.